apps/api_web/test/api_web/params_test.exs

```diff
@@ -54,17 +54,46 @@ defmodule ApiWeb.ParamsTest do
         }
       }
 
-      {:ok, %{params: params}}
+      {:ok, %{params: params, conn: conn}}
     end
 
-    test "filters items from 2nd arg", %{params: params} do
-      assert Params.filter_params(params, ["route", "trip", "stop"]) ==
+    test "filters items from 2nd arg", %{params: params, conn: conn} do
+      assert Params.filter_params(params, ["route", "trip", "stop"], conn) ==
                {:ok,
                 %{
                   "route" => "1,2,3",
                   "trip" => "1,2,3",
                   "stop" => "4,5,6"
                 }}
     end
+
+    test "returns error in case of unsupported filter", %{params: params, conn: conn} do
+      assert Params.filter_params(params, ["route"], conn) == {:error, :bad_filter, ~w(stop trip)}
+
+      conn = assign(conn, :api_version, "2019-02-12")
+      assert Params.filter_params(params, ["route"], conn) == {:ok, %{"route" => "1,2,3"}}
+    end
+  end
+
+  describe "validate_includes/2" do
+    test "returns ok for valid includes", %{conn: conn} do
+      assert Params.validate_includes(%{"include" => "stops,trips"}, ~w(stops routes trips), conn) ==
+               {:ok, ~w(stops trips)}
+    end
+
+    test "returns error for invalid includes", %{conn: conn} do
+      assert Params.validate_includes(%{"include" => "stops,routes"}, ~w(stops trips), conn) ==
+               {:error, :bad_include, ~w(routes)}
+
+      conn = assign(conn, :api_version, "2019-02-12")
+
```

Comment:
Same with these: two separate tests so that it's more clear what's being tested.

Reply:
Fixed